react-exercises/modal-react/src/component/LoginModal.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { Component } from "react";
+import "./modal.css";
+
+export default class LoginModal extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      ...props,
+      username: "",
+      password: "",
+    };
+    this.handleFormSubmit = this.handleFormSubmit.bind(this);
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.closeModal = this.closeModal.bind(this);
+  }
+
+  /**
+   * The method will bind the react component state with input
+   * State of HTML(User Input.)
+   * @param event {Event} Event A javascript Event Object
+   */
+  handleInputChange(event) {
+    const newState = this.state;
+    newState[event.target.name] = event.target.value;
+    this.setState(newState);
+  }
+
+  /**
+   * The method will override the default behavior of a
+   * form on submission and will handle form submission.
+   * @param event {Event} Event A javascript Event Object
+   */
+  handleFormSubmit(event) {
+    event.preventDefault();
+    const { username, password } = this.state;
+    if (username === "sonutj" && password === "password") {
+      alert("Login Successfull");
+    } else {
+      alert("Invalid username or password");
+    }
+    this.setState({
+      ...this.state,
+      password: "",
+    });
+  }
+
+  /**
+   * Close modal will handle the hiding o the modal when clicked on close button.
+   * @param event {Event} Event A javascript Event Object
+   */
+  closeModal(event) {
+    this.state.onClose();
+  }
+
+  render() {
+    const showHideModal = this.state.status
```

Comment:
you can write like below to call status from props
const showHideModal=this.sprops.status